tools/issue-labeler/src/SearchIndexCreator/IssueTriageContentRetrieval.cs

```diff
@@ -0,0 +1,370 @@
+            CodeOwnerUtils.codeOwnersFilePathOverride = codeownersContents[0].DownloadUrl;
+
+            var issues = await RetrieveAllIssuesAsync(repoOwner);
+            var documents = await RetrieveAllDocumentsAsync(repoOwner);
+
+            var results = new List<IssueTriageContent>();
+            results.AddRange(issues);
+            results.AddRange(documents);
+
+            return results;
+        }
+
+        public async Task UploadSearchContent(List<IssueTriageContent> IssueTriageContents)
+        {
+            if (string.IsNullOrEmpty(_accountName))
+            {
+                throw new ArgumentNullException(nameof(_accountName), "Storage account name cannot be null or empty.");
+            }
+            var blobServiceClient = GetBlobServiceClient(_accountName);
+            await EnsureBlobSoftDeleteEnabled(blobServiceClient);
+
+            var containerClient = blobServiceClient.GetBlobContainerClient(_containerName);
+            await containerClient.CreateIfNotExistsAsync();
+
+            foreach (var content in IssueTriageContents)
+            {
+                var blobClient = containerClient.GetBlobClient(content.Id);
+                try
+                {
+                    var jsonContent = JsonConvert.SerializeObject(content);
+                    var blobHttpHeaders = new Azure.Storage.Blobs.Models.BlobHttpHeaders
+                    {
+                        ContentType = "application/json"
+                    };
+                    using (var stream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(jsonContent)))
+                    {
+                        await blobClient.UploadAsync(stream, new Azure.Storage.Blobs.Models.BlobUploadOptions
+                        {
+                            HttpHeaders = blobHttpHeaders
+                        });
+                    }
+                    Console.WriteLine($"Uploaded {content.Id} to {containerClient.Name} container.");
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception($"Error uploading file: {content.Id}", ex);
+                }
+            }
+        }
+
+        public async Task<List<IssuePayload>> RetrieveIssueExamples(string repoOwner, string repoName, int days)
+        {
+            var issueReq = new RepositoryIssueRequest
+            {
+                State = ItemStateFilter.Closed,
+                Filter = IssueFilter.All,
+                Since = DateTimeOffset.Now.Subtract(TimeSpan.FromDays(days))
+            };
+            issueReq.Labels.Add("customer-reported");
+            issueReq.Labels.Add("issue-addressed");
+
+            var issues = await _client.Issue.GetAllForRepository(repoOwner, repoName, issueReq);
+            var results = new List<IssuePayload>();
+
+            foreach (var issue in issues)
+            {
+                if (issue.PullRequest != null)
```

Comment:
We also need to be sure that we have exactly 1 service label (pink) and 1 category label (yellow) on the issue before adding it.  The color is significant in our classification.    

Pink:  `#e99695`
Yellow: `#ffeb77`

If the issue has a pink, a yellow, "customer-reported" and "issue-addressed" then it's good to include in the KB.